### PR TITLE
Simplify mirror commands, messages

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Simplifies admin commands and net messages. Precursor to #7179.
- Namespaces `VariableViewer`.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 1]
```diff
@@ -371,13 +373,14 @@ public void CmdRegisterVote(bool isFor)
 	}
 
 	[Command]
-	public void CmdVetoRestartVote(string adminId, string adminToken)
+	public void CmdVetoRestartVote(NetworkConnectionToClient sender = null)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
	public void CmdVetoRestartVote()
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 2]
```diff
@@ -371,13 +373,14 @@ public void CmdRegisterVote(bool isFor)
...
 	{
-		var admin = PlayerList.Instance.GetAdmin(adminId, adminToken);
-		if (admin == null) return;
+		if (AdminCommandsManager.IsAdmin(sender, out var player))
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
		if (AdminCommandsManager.IsAdmin(connectionToClient, out var player))
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 3]
```diff
@@ -530,17 +533,12 @@ public void CmdCommitSuicide()
 	// Respawn action for Deathmatch v 0.1.3
 
 	[Command]
-	public void CmdRespawnPlayer(string adminID, string adminToken)
+	public void CmdRespawnPlayer(NetworkConnectionToClient sender = null)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
	public void CmdRespawnPlayer()
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 4]
```diff
@@ -530,17 +533,12 @@ public void CmdCommitSuicide()
...
 	{
-		if (GameManager.Instance.RespawnCurrentlyAllowed ||
-		    PlayerList.Instance.GetAdmin(adminID, adminToken))
+		if (AdminCommandsManager.IsAdmin(sender, out var player) || GameManager.Instance.RespawnCurrentlyAllowed)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
		if (AdminCommandsManager.IsAdmin(connectionToClient, out var player) || GameManager.Instance.RespawnCurrentlyAllowed)
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 5]
```diff
@@ -848,17 +846,17 @@ public void ServerGhostPerformTeleport(Vector3 s3)
 	#region Admin-only
 
 	[Command]
-	public void CmdAGhost(string adminId, string adminToken)
+	public void CmdAGhost(NetworkConnectionToClient sender = null)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
	public void CmdAGhost()
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 6]
```diff
@@ -848,17 +846,17 @@ public void ServerGhostPerformTeleport(Vector3 s3)
...
 	{
-		ServerAGhost(adminId, adminToken);
+		if (AdminCommandsManager.IsAdmin(sender, out _))
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
		if (AdminCommandsManager.IsAdmin(connectionToClient, out _))
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 7]
```diff
@@ -872,10 +870,10 @@ public void ServerAGhost(string adminId, string adminToken)
 	}
 
 	[Command]
-	public void CmdAdminMakeHotspot(GameObject onObject, string adminId, string adminToken)
+	public void CmdAdminMakeHotspot(GameObject onObject, NetworkConnectionToClient sender = null)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
	public void CmdAdminMakeHotspot(GameObject onObject)
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 8]
```diff
@@ -872,10 +870,10 @@ public void ServerAGhost(string adminId, string adminToken)
...
 	{
-		var admin = PlayerList.Instance.GetAdmin(adminId, adminToken);
-		if (admin == null) return;
+		if (AdminCommandsManager.IsAdmin(sender, out _) == false) return;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
		if (AdminCommandsManager.IsAdmin(connectionToClient, out _) == false) return;
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 9]
```diff
@@ -888,10 +886,9 @@ public void CmdAdminMakeHotspot(GameObject onObject, string adminId, string admi
 	}
 
 	[Command]
-	public void CmdAdminSmash(GameObject toSmash, string adminId, string adminToken)
+	public void CmdAdminSmash(GameObject toSmash, NetworkConnectionToClient sender = null)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
	public void CmdAdminSmash(GameObject toSmash)
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 10]
```diff
@@ -888,10 +886,9 @@ public void CmdAdminMakeHotspot(GameObject onObject, string adminId, string admi
...
 	{
-		var admin = PlayerList.Instance.GetAdmin(adminId, adminToken);
-		if (admin == null) return;
+		if (AdminCommandsManager.IsAdmin(sender, out _) == false) return;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
		if (AdminCommandsManager.IsAdmin(connectionToClient, out _) == false) return;
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 11]
```diff
@@ -908,9 +905,12 @@ public void CmdAdminSmash(GameObject toSmash, string adminId, string adminToken)
 	}
 
 	[Command]
-	public void CmdGetAdminOverlayFullUpdate(string adminId, string adminToken)
+	public void CmdGetAdminOverlayFullUpdate(NetworkConnectionToClient sender = null)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
	public void CmdGetAdminOverlayFullUpdate()
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 12]
```diff
@@ -908,9 +905,12 @@ public void CmdAdminSmash(GameObject toSmash, string adminId, string adminToken)
...
 	{
-		AdminOverlay.RequestFullUpdate(adminId, adminToken);
+		if (AdminCommandsManager.IsAdmin(sender, out var player))
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
		if (AdminCommandsManager.IsAdmin(connectionToClient, out var player))
```

**Comment by ThatDan123** [comment]
Can just use `connectionToClient` to get the sender when in PlayerNetworkActions as its on the player object

File: UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs [hunk 13]
```diff
@@ -54,14 +54,13 @@ private void Awake()
 		/// <param name="adminToken"></param>
 		/// <param name="sender">The client which sends the command, this is populated by mirror so doesnt need to be manually
 		/// put in the parameters when calling the commands</param>
-		public static bool IsAdmin(string adminId, string adminToken, NetworkConnectionToClient sender)
+		public static bool IsAdmin(NetworkConnectionToClient sender, out ConnectedPlayer player)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs:54]:
```suggestion
		public static bool IsAdmin(NetworkConnection sender, out ConnectedPlayer player)
```

Should fix the tests

**Review by ThatDan123 — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]
I'm going to be sorting out a lot of merge conflicts of my stuff is ok xD

**Comment by Bod9001** [comment]
wait for gillsto see if it matches the bounty

**Pull request merged**